controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -0,0 +1,171 @@
+    } else if (command == "DEV_SET_CONFIG") {
+        return DEV_SET_CONFIG;
+    }
+
+    return WFA_CA_COMMAND_MAX;
+}
+
+void wfa_ca::reply_invalid_error(Socket *sd, ieee1905_1::CmduMessageTx &cmdu_tx,
+                                 eWfaCaStatus status, const std::string &description)
+{
+    if (status != INVALID || status != ERROR) {
+        LOG(WARNING) << "function misuse, status is not INVALID or ERROR";
+        return;
+    }
+
+    std::string reply_string = "status," + wfa_ca_status(status) + ",errorCode," + description;
+    reply(sd, cmdu_tx, reply_string, false);
+}
+
+bool wfa_ca::reply(Socket *sd, ieee1905_1::CmduMessageTx &cmdu_tx, const std::string &reply_string,
+                   bool pending_second_response)
+{
+    auto response =
+        message_com::create_vs_message<beerocks_message::cACTION_BML_WFA_CA_CONTROLLER_RESPONSE>(
+            cmdu_tx);
+
+    if (!response) {
+        LOG(ERROR) << "Failed building cACTION_BML_WFA_CA_CONTROLLER_RESPONSE message!";
+        return false;
+    }
+
+    response->alloc_reply(reply_string.size() + 1);
+    auto reply_tuple = response->reply(0);
+    if (!std::get<0>(reply_tuple)) {
+        LOG(ERROR) << "getting reply entry has failed!";
+        return false;
+    }
+
+    char *reply_ptr = (char *)(&std::get<1>(reply_tuple));
+    string_utils::copy_string(reply_ptr, reply_string.c_str(), reply_string.size() + 1);
+    response->pending_second_response() = pending_second_response;
+    return message_com::send_cmdu(sd, cmdu_tx);
+}
+
+void wfa_ca::handle_wfa_ca_message(
+    Socket *sd, std::shared_ptr<beerocks_message::cACTION_HEADER> beerocks_header,
+    ieee1905_1::CmduMessageRx &cmdu_rx, ieee1905_1::CmduMessageTx &cmdu_tx, db &database,
+    task_pool &tasks)
+{
+    auto request = cmdu_rx.addClass<beerocks_message::cACTION_BML_WFA_CA_CONTROLLER_REQUEST>();
+
+    if (!request) {
+        LOG(ERROR) << "addClass cACTION_BML_WFA_CA_CONTROLLER_REQUEST failed";
+        return;
+    }
+
+    auto command_tuple = request->command(0);
+    if (!std::get<0>(command_tuple)) {
+        LOG(ERROR) << "getting command entry has failed!";
+        return;
+    }
+    const char *command_ptr = (char *)(&std::get<1>(command_tuple));
+
+    LOG(DEBUG) << "Received ACTION_BML_WFA_CA_CONTROLLER_REQUEST, command=" << command_ptr;
+
+    std::string command(command_ptr);
+    auto cmd_tokens_vec = string_utils::str_split(command_ptr, ',');
+
+    if (cmd_tokens_vec.size() < 1) {
+        LOG(ERROR) << "Invalid WFA-CA command";
+        reply_invalid_error(sd, cmdu_tx, eWfaCaStatus::INVALID, "empty command");
+    }
+
+    auto command_type     = wfa_ca_command(cmd_tokens_vec[0]);
+    auto command_type_str = cmd_tokens_vec[0];
+    cmd_tokens_vec.erase(cmd_tokens_vec.begin()); // erase command token
+
+    switch (command_type) {
+    case eWfaCaCommand::CA_GET_VERSION:
+        /* code */
+        break;
+    case eWfaCaCommand::DEV_CONFIGURE_IE:
+        /* code */
```

Comment:
I prefer a comment starting with a `TODO` for each of these, with a short explanation on what should be done if it isn't obvious (nitpick)

Reply:
VSCODE add automatically /* code */
I changed it, but didn't add an explanation.
It is obvious that what needs to be done is to handle the command...